libcudacxx/test/libcudacxx/cuda/bit/bitfield.pass.cpp

```diff
@@ -0,0 +1,79 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the libcu++ Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#include <cuda/bit>
+#include <cuda/std/cassert>
+#include <cuda/std/cstdint>
+#include <cuda/std/type_traits>
+
+#include "test_macros.h"
+
+template <typename T>
+__host__ __device__ constexpr bool test()
+{
+  using nl             = cuda::std::numeric_limits<T>;
+  constexpr T all_ones = static_cast<T>(~T{0});
+  unused(all_ones);
+  assert(cuda::bitfield_insert(T{0}, all_ones, 0, 1) == 1);
+  assert(cuda::bitfield_insert(T{0}, all_ones, 1, 1) == 0b10);
+  assert(cuda::bitfield_insert(T{0b10}, all_ones, 0, 1) == 0b11);
+  assert(cuda::bitfield_insert(all_ones, all_ones, 0, 1) == all_ones);
+  assert(cuda::bitfield_insert(all_ones, all_ones, 2, 1) == all_ones);
```

Comment:
How about changing this test case to write a `0`-bit at position two to increase coverage.